# Project: Analyze International Debt Statistics

This project analyzes international debt data collected by The World Bank. The dataset includes debt indicators for developing countries across various categories. Using SQL, I extracted key insights to address three primary questions:

- What is the number of distinct countries present in the database?
- What country has the highest amount of debt?
- What country has the lowest amount of repayments?

Below is a description of the table:

**`international_debt` table**

| Column | Definition | Data Type |
|-|-|-|
|country_name|Name of the country|`varchar`|
|country_code|Code representing the country|`varchar`|
|indicator_name|Description of the debt indicator|`varchar`|
|indicator_code|Code representing the debt indicator|`varchar`|
|debt|Value of the debt indicator for the given country (in current US dollars)|`float`|

**Counting Distinct Countries**

Objective: To determine the diversity of countries represented in the dataset.

In [32]:
-- What is the number of distinct countries present?
SELECT COUNT (DISTINCT country_code) as total_distinct_countries
FROM international_debt;

total_distinct_countries
0                       124

Result: The dataset contains 124 unique countries.

Analysis: With 124 countries, the dataset provides a broad view of international debt, making it a reliable source for understanding global trends and identifying key insights about developing economies.

**Identifying the Country with the Highest Debt**

Objective: To find the country with the largest debt in the dataset.

In [33]:
-- What country has the highest amount of debt?
SELECT country_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 1;

country_name    total_debt
0        China  2.857935e+11

Result: China has the highest total debt amounting to $285.79 billion USD.


Analysis: China’s significant debt is likely tied to its massive investments in infrastructure and economic expansion. This finding reflects the scale of its financial operations and its impact on global lending dynamics.

**Finding the Country with the Lowest Principal Repayments**
Objective: Determine which country is making the smallest principal repayments on long-term external debt.

In [34]:
-- What country has the lowest amount of repayments?
SELECT country_name, indicator_name, MIN(debt) AS lowest_repayment
FROM international_debt
WHERE indicator_code = 'DT.AMT.DLXF.CD'
GROUP BY country_name, indicator_name
ORDER BY lowest_repayment
LIMIT 1;

country_name  ... lowest_repayment
0  Timor-Leste  ...           825000

[1 rows x 3 columns]

Result: Timor-Leste has the lowest repayment at $825,000 USD.


Analysis: Timor-Leste’s low repayment amount suggests it might have fewer outstanding debts or be in the early stages of repayment. It highlights how smaller nations manage their debt obligations compared to larger economies.